unified_planning/plans/ttp_to_stn.py

```diff
@@ -0,0 +1,270 @@
+                )
+                self.table[action_cpl].update({effect_time: (pconditions[1], effects)})
+        return self.table
+
+    def table_to_events(self) -> List[Tuple[Fraction, InstantaneousAction]]:
+        """
+        Return a list where each time is paired with an Instantaneous Action.
+        Each Instantaneous Action is created from preconditions and effects from get_table_event()
+        """
+        self.get_table_event()
+        for action_cpl, c_e_dict in self.table.items():
+            start = action_cpl[0]
+            duration = action_cpl[2]
+            action = action_cpl[1]
+            if duration is None:
+                assert isinstance(
+                    action.action, InstantaneousAction
+                ), "Error, None duration specified for non InstantaneousAction"
+                continue
+            for time_pt, cond_eff_couple in c_e_dict.items():
+
+                time = _absolute_time(time_pt, start, duration)
+                inst_action = InstantaneousAction(str(action) + str(time_pt))
+
+                # set conditions and effects to each Instantaneous Action
+                if len(cond_eff_couple[0]) > 0:
+                    for cond in cond_eff_couple[0]:
+                        inst_action.add_precondition(cond)
+
+                if len(cond_eff_couple[1]) > 0:
+                    for effect in cond_eff_couple[1]:
+                        inst_action.effects.append(effect)
+
+                self.events += [(time, inst_action)]
+        return self.events
+
+    def sort_events(self) -> List[Tuple[Fraction, InstantaneousAction]]:
+        """
+        Return the table of instant action sorted by time.
+        """
+        self.table_to_events()
+
+        self.events_sorted = sorted(self.events, key=lambda acts: acts[0])
+
+        print("\r\nAFTER SORT")
+        for tim, act in self.events_sorted:
+            print("%s: %s" % (float(tim), act._name))
+        return self.events_sorted
+
+    def events_to_partial_order_plan(self):
+        """
+        Return the partial order plan from the list of events sorted.
+        """
+        self.sort_events()
+        list_act = [ActionInstance(i[1]) for i in self.events_sorted]
+        seqplan = SequentialPlan(list_act)
+        partial_order_plan = seqplan._to_partial_order_plan(self.problem)
+        self.partial_order_plan = partial_order_plan
+        print("\r\npartial order plan ", partial_order_plan)
+        return partial_order_plan
+
+    def partial_order_plan_to_stn(self) -> nx.DiGraph:
+        """
+        Stn with Start and End nodes
+        Return the stn from the partial order plan.
+        """
+        graph = self.partial_order_plan._graph
+
+        for ai_current, l_next_ai in self.partial_order_plan.get_adjacency_list.items():
+            ai_current_time = [
+                item[0] for item in self.events_sorted if item[1] == ai_current.action
+            ][0]
```

Comment:
Because of adjacency list is a dict of ActionInstance and not a dict of Instantaneous action in the other way around I need to cross the adj list also.